### ISSUE TITLE
Ownership links on user pages linking to a catalog page without the expected items

### ISSUE THREAD

**Issue by vierno** [backstage/backstage#7254]
Recently was introduced a new navigation from the `OwnershipCard` to the catalog table with filters, but when it's used from an user profile, it will result unexpected results (I see that the user owns X services, but when I click the box, the table shows zero results).

## Expected Behavior

When clicking on the "Ownership" boxes on an user page, the catalog page should list the entities owned by the user, including those that are owned by the user's groups.

## Current Behavior

Clicking on the "Ownership" links from an user page, the catalog will be filtered `user:<user>` but that does not show the same number of results that are accounted on the previous page.

## Possible Solution

Include filters with all the user's groups on the link or the `user:<user>` filter on the catalog could include entities owned by the users' groups.

## Steps to Reproduce

On an user's page (e.g. https://demo.backstage.io/catalog/default/user/breanna.davison), click on any of the "Ownership" boxes and the next page won't have the expected results.

## Your Environment

Tested with chrome on macOS.

**Comment by Rugvip** [comment]
Yeah that doesn't look right! I think the proposed solution of including all the groups in the filter would be the way to go. It's the less exciting fix since changing the behaviours of the filters could have some broader effects 😅
